### PR TITLE
Adding SystemClockFromEpoch to be able to set time from fixed time base

### PR DESCRIPTION
**Pull request by jleskovar**
additional Clock implementation that lets you set time from a fixed epoch

### CODE REVIEW

**Comment by jleskovar** [comment]
closing for now -- accidentally left in compile error :s

**Comment by pron** [comment]
And why not make it offset/set the epoch for any given clock, with `SystemClock` being the default? See [`ScaledClock`](https://github.com/puniverse/timewarp/blob/master/src/main/java/co/paralleluniverse/vtime/ScaledClock.java)

**Comment by jleskovar** [comment]
Alright, re-opened with the compile fix, and made it more generic.

File: src/main/java/co/paralleluniverse/vtime/FixedEpochClock.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    public FixedEpochClock(Clock baseClock, long epoch) {
+        this.baseClock = baseClock;
+        this.offset = epoch - baseClock.currentTimeMillis();
+    }
+
+    @Override
+    public String toString() {
+        return "FixedEpochClock";
```
Comment (pron) [inline src/main/java/co/paralleluniverse/vtime/FixedEpochClock.java]: Mention the base clock

File: src/main/java/co/paralleluniverse/vtime/FixedEpochClock.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    @Override
+    long System_nanoTime() {
+        return baseClock.nanoTime() + (offset * 1_000_000);
+    }
+
+    @Override
+    void Object_wait(Object obj, long timeout) throws InterruptedException {
+        obj.wait(timeout);
```
Comment (pron) [inline src/main/java/co/paralleluniverse/vtime/FixedEpochClock.java]: use `baseClock.Object_wait(obj, timeout)`

File: src/main/java/co/paralleluniverse/vtime/FixedEpochClock.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    @Override
+    void Thread_sleep(long millis) throws InterruptedException {
+        Thread.sleep(millis);
```
Comment (pron) [inline src/main/java/co/paralleluniverse/vtime/FixedEpochClock.java]: Use `baseClock`

File: src/main/java/co/paralleluniverse/vtime/FixedEpochClock.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    @Override
+    void Unsafe_park(sun.misc.Unsafe unsafe, boolean isAbsolute, long timeout) {
+        park(unsafe, isAbsolute, timeout);
```
Comment (pron) [inline src/main/java/co/paralleluniverse/vtime/FixedEpochClock.java]: Use `baseClock`.

**Comment by pron** [comment]
Thanks! Please fix the issues I commented on and I'll merge the PR.

**Comment by jleskovar** [comment]
Changes made, please review, 
cheers.

**Comment by pron** [comment]
Very good. Thank you!

**Pull request merged**